[docs] proposed changes to coin_registry (#24507)
MystenLabs/sui: docs/content/guides/developer/currency.mdx

The registry system supports two currency creation methods:

1. **Standard creation:** Call `new_currency<T>()` when creating the coin outside of the `init` function of your package.

> 1. **OTW creation:** Use `new_currency_with_otw<T>()` with a One-Time Witness for uniqueness proof. Note: that for proper creation and RPC support it requires a second transaction to promote the currency to the registry.

Reviewer: ```suggestion
1. **OTW creation:** Use `new_currency_with_otw<T>()` with a One-Time Witness for uniqueness proof.

:::danger 
Proper creation and RPC support **requires** a second transaction to promote the currency to the registry.
:::
```

:::caution Important

> After initialization of a currency using the OTW method, you must call [`coin_registry::finalize_registration`](./../../standards/currency.mdx#coin-finalization) to create the shared `Currency` object that the Coin Registry can track.

Reviewer: ```suggestion
After initialization of a currency using the OTW method, you **must** call [`coin_registry::finalize_registration`](/standards/currency.mdx#coin-finalization) to create the shared `Currency` object that the Coin Registry can track.
```
